Sources/MapboxMaps/Foundation/BaseMapView.swift

```diff
@@ -3,100 +3,40 @@
 import UIKit
 import Turf
 
-// swiftlint:disable file_length type_body_length
-
-public enum PreferredFPS: RawRepresentable, Equatable {
-
-    /**
-     Create a `PreferredFPS` value from an `Int`.
-     - Parameter rawValue: The `Int` value to use as the preferred frames per second.
-     */
-    public init?(rawValue: Int) {
-        switch rawValue {
-        case Self.lowPower.rawValue:
-            self = .lowPower
-        case Self.normal.rawValue:
-            self = .normal
-        case Self.maximum.rawValue:
-            self = .maximum
-        default:
-            self = .custom(fps: rawValue)
-        }
-    }
-
-    public typealias RawValue = Int
-
-    /// The default frame rate. This can be either 30 FPS or 60 FPS, depending on
-    /// device capabilities.
-    case normal
-
-    /// A conservative frame rate; typically 30 FPS.
-    case lowPower
-
-    /// The maximum supported frame rate; typically 60 FPS.
-    case maximum
-
-    /// A custom frame rate. The default value is 30 FPS.
-    case custom(fps: Int)
-
-    /// The preferred frames per second as an `Int` value.
-    public var rawValue: Int {
-        switch self {
-        case .lowPower:
-            return 30
-        case .normal:
-            return -1
-        case .maximum:
-            return 0
-        case .custom(let fps):
-            // TODO: Check that value is a valid FPS value.
-            return fps
-        }
-    }
-
-}
-
-open class ObserverConcrete: Observer {
-
-    /// Map of event types to subscribed event handlers
-    internal var eventHandlers: [String: [(MapboxCoreMaps.Event) -> Void]] = [:]
-
-    /// Notify correct handler
-    public func notify(for event: MapboxCoreMaps.Event) {
-        let handlers = eventHandlers[event.type]
-        handlers?.forEach({ (handler) in
-            handler(event)
-        })
-    }
-}
+// swiftlint:disable file_length
 
 internal typealias PendingAnimationCompletion = (completion: AnimationCompletion, animatingPosition: UIViewAnimatingPosition)
 
-open class BaseMapView: UIView, MapClient, MBMMetalViewProvider, CameraViewDelegate {
+open class BaseMapView: UIView, CameraViewDelegate {
 
     /// The underlying renderer object responsible for rendering the map
-    public var __map: Map!
+    public private(set) var __map: Map!
+
+    private let mapClient = DelegatingMapClient()
+    private let observer = DelegatingObserver()
 
     /// The underlying metal view that is used to render the map
-    internal var metalView: MTKView?
+    internal private(set) var metalView: MTKView?
```

Comment:
if this is internal, what value does a private(set) give us?

Reply:
It's a way of saying that only the BaseMapView is allowed to modify its metalView. The same principles we use when designing public APIs can be valuable when designing the internal APIs as well.